src/nodedef.cpp

```diff
@@ -213,10 +214,24 @@ void TileDef::serialize(std::ostream &os, u16 protocol_version) const
 		// Before f018737, TextureSource::getTextureAverageColor did not handle
 		// missing textures. "[png" can be used as base texture, but is not known
 		// on older clients. Hence use "blank.png" to avoid this problem.
-		if (!name.empty() && name[0] == '[')
-			os << serializeString16("blank.png^" + name);
-		else
+		// Note that modifiers such as [combine can be used as base textures
+		// and are compatible with older clients, so we could specifically single out [png:,
+		// but this would require the list here to be updated whenever a new base modifier
+		// is introduced. Instead, we whitelist the three cases that are supported by the 
+		// older clients and prefix anything else
+		bool pass_through = true;
+
+		// only check prefixes if the name begins with a [
```

Comment:
redundant comment

Reply:
I actaully expected an objection to the tune that the extra checks for length > 0 and [ as first character are superfluous, and was tempted to extend the comment to include that this is a perf optimization to only do the later (slower) checks if necessary. but I can remove it altogether.